x/mongo/driver/operation.go

```diff
@@ -333,6 +333,14 @@ func (op Operation) Execute(ctx context.Context, scratch []byte) error {
 		}
 	}
 
+	// Use RetryContext (retry indefinitely until context errors) if context is a Timeout context and
+	// RetryMode is not already RetryChangeStream (change streams handle their own retryability when
+	// the context is a Timeout context).
+	if internal.IsTimeoutContext(ctx) && (op.RetryMode == nil || *op.RetryMode != RetryChangeStream) {
+		rm := RetryContext
+		op.RetryMode = &rm
```

Comment:
Changing the value of an enum at runtime is probably not a great idea, it would change the value of the underlying structure which, for fields pointing to enums, [should probably be static](https://stackoverflow.com/a/20168577). What are your thoughts on just extending the switch command like this:

```go
case RetryChangeStream:
    if internal.IsTimeoutContext(ctx) {
        retries = -1
   }
}
```
For the nil case:
```go
if internal.IsTimeoutContext(ctx) && op.RetryMode == nil {
  retries = -1
}
```

Reply:
Since I've removed `RetryChangeStream`, this now only sets the `op.RetryMode` when it is `nil`. Do you still have the same concern about changing `nil` to `&RetryContext`? I suppose we're still modifying the `Operation` object at runtime.

Reply:
I think it is still concerning. Instead of updating user-defined data on the operation object, you can accomplish the same functionality by extending this branch:

```go
if op.RetryMode != nil {
} else {
  // your new logic
}
```

Reply:
I've now changed the logic to not modify the `RetryMode` enum value and instead just modify the `retries` number to `-1`.